### PR TITLE
Smart bullets

### PR DESCRIPTION
**Pull request by kostaldavid8**
When you hit enter on a line with a bullet, you get a new one on the new line.
Also when you hit tab after a bullet, it automatically indents.
It makes typing with bullets much more pleasant.

### CODE REVIEW

File: browser/components/CodeEditor.js [hunk 1]
```diff
@@ -57,17 +57,35 @@ export default class CodeEditor extends React.Component {
...
-            if (cm.getOption('indentWithTabs')) {
-              cm.execCommand('insertTab')
+            let tabs = cm.getOption('indentWithTabs')
+            if (line.trimLeft() === '- ' || line.trimLeft() === '* ') {
+              cm.execCommand('goLineStart')
+              if (tabs) cm.execCommand('insertTab'); else cm.execCommand('insertSoftTab')
+              cm.execCommand('goLineEnd')
             } else {
-              cm.execCommand('insertSoftTab')
+              if (tabs) cm.execCommand('insertTab'); else cm.execCommand('insertSoftTab')
```
Comment (sota1235) [inline browser/components/CodeEditor.js]: Please use `{}` statement. It's more readable.

```javascript
if () {
} else {
}
```

File: browser/components/CodeEditor.js [hunk 2]
```diff
@@ -57,17 +57,35 @@ export default class CodeEditor extends React.Component {
...
+          let cursor = cm.getCursor()
+          let line = cm.getLine(cursor.line)
           if (cm.somethingSelected()) cm.indentSelection('add')
           else {
-            if (cm.getOption('indentWithTabs')) {
-              cm.execCommand('insertTab')
+            let tabs = cm.getOption('indentWithTabs')
+            if (line.trimLeft() === '- ' || line.trimLeft() === '* ') {
+              cm.execCommand('goLineStart')
+              if (tabs) cm.execCommand('insertTab'); else cm.execCommand('insertSoftTab')
```
Comment (sota1235) [inline browser/components/CodeEditor.js]: Same as https://github.com/BoostIO/Boostnote/pull/289/files#r101394868

File: browser/components/CodeEditor.js [hunk 3]
```diff
@@ -57,17 +57,35 @@ export default class CodeEditor extends React.Component {
       inputStyle: 'textarea',
       extraKeys: {
         Tab: function (cm) {
+          let cursor = cm.getCursor()
+          let line = cm.getLine(cursor.line)
```
Comment (sota1235) [inline browser/components/CodeEditor.js]: Please use `const` if you wouldn't reassign the value to variable.

File: browser/components/CodeEditor.js [hunk 4]
```diff
@@ -57,17 +57,35 @@ export default class CodeEditor extends React.Component {
...
             } else {
-              cm.execCommand('insertSoftTab')
+              if (tabs) cm.execCommand('insertTab'); else cm.execCommand('insertSoftTab')
             }
           }
         },
         'Cmd-T': function (cm) {
           // Do nothing
+        },
+        Enter: function (cm) {
```
Comment (sota1235) [inline browser/components/CodeEditor.js]: Please use arrow function. We can use ES2015 :)

**Comment by sota1235** [comment]
@kostaldavid8
Thank you for your PR.
Please check my comments :D

**Comment by kostaldavid8** [comment]
I hope it's OK now :)

**Comment by kostaldavid8** [comment]
Added auto-increment for numbered lists

**Comment by sota1235** [comment]
I use this function at local and I felt so good! But I want to fix one more point.
We can write list with using `*` and `-` you know. And you can also write list with using `+`!

So can you support `+` mark with this function?

Thank you :)

**Comment by kostaldavid8** [comment]
Done :)

**Comment by kostaldavid8** [comment]
What is the problem with CI?

**Comment by asmsuechan** [comment]
@kostaldavid8 
Occasionally, this test fails :sob:. So I retried.

**Comment by sota1235** [comment]
Good fixing. Thanks a lot!

**Pull request merged**